packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/log_query_and_deprecation.ts

```diff
@@ -58,35 +125,43 @@ function getContentLength(headers?: IncomingHttpHeaders): number | undefined {
  *
  * so it could be copy-pasted into the Dev console
  */
-function getResponseMessage(event: DiagnosticResult, bytesMsg: string): string {
-  const errorMeta = getRequestDebugMeta(event);
-  const body = errorMeta.body ? `\n${errorMeta.body}` : '';
-  return `${errorMeta.statusCode}${bytesMsg}\n${errorMeta.method} ${errorMeta.url}${body}`;
+function getResponseMessage(
+  event: DiagnosticResult,
+  bytesMsg: string,
+  apisToRedactInLogs: ElasticsearchPotentiallyLeakingApi[]
+): string {
+  const debugMeta = getRequestDebugMeta(event, apisToRedactInLogs);
+  const body = debugMeta.body ? `\n${debugMeta.body}` : '';
+  return `${debugMeta.statusCode}${bytesMsg}\n${debugMeta.method} ${debugMeta.url}${body}`;
 }
 
 /**
  * Returns stringified debug information from an Elasticsearch request event
  * useful for logging in case of an unexpected failure.
  */
-export function getRequestDebugMeta(event: DiagnosticResult): {
-  url: string;
-  body: string;
-  statusCode: number | null;
-  method: string;
-} {
+export function getRequestDebugMeta(
+  event: DiagnosticResult,
+  apisToRedactInLogs?: ElasticsearchPotentiallyLeakingApi[]
```

Comment:
Mind that this method is also used for the Saved Objects migration logger. This is the reason it's optional here. 
I doubt it'll ever use any potentially leaking API that we might need to extend from the config (if it's part of the const above, there's no risk).

In the future, if this becomes a risk, we can share the config with the SO Migrator service to provide the config extension.